core/modules/replica/FindAllJob.h

```diff
@@ -0,0 +1,258 @@
+    ///               all database-specific chunks are also complete (healthy).
+    ///
+    ///      [chunk][worker]
+    ///
+    std::map<unsigned int,
+             std::map<std::string,
+                      bool>> isGood;
+};
+
+/**
+  * Class FindAllJob represents a tool which will find all replicas
+  * of all chunks on all worker nodes.
+  */
+class FindAllJob
+    :   public Job  {
+
+public:
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<FindAllJob> Ptr;
+
+    /// The function type for notifications on the completon of the request
+    typedef std::function<void(Ptr)> CallbackType;
+
+    /// @return default options object for this type of a request
+    static Job::Options const& defaultOptions();
+
+    /**
+     * Static factory method is needed to prevent issue with the lifespan
+     * and memory management of instances created otherwise (as values or via
+     * low-level pointers).
+     *
+     * @param databaseFamily  - name of a database family
+     * @param saveReplicaInfo - save replica info in a database
+     * @param controller      - for launching requests
+     * @param parentJobId     - optional identifier of a parent job
+     * @param onFinish        - callback function to be called upon a completion of the job
+     * @param options         - job options
+     *
+     * @return pointer to the created object
+     */
+    static Ptr create(std::string const& databaseFamily,
+                      bool saveReplicaInfo,
+                      Controller::Ptr const& controller,
+                      std::string const& parentJobId,
+                      CallbackType onFinish,
+                      Job::Options const& options=defaultOptions());
+
+    // Default construction and copy semantics are prohibited
+
+    FindAllJob() = delete;
+    FindAllJob(FindAllJob const&) = delete;
+    FindAllJob& operator=(FindAllJob const&) = delete;
+
+    ~FindAllJob() final = default;
+
+    /// @return the name of a database family defining a scope of the operation
+    std::string const& databaseFamily() const { return _databaseFamily; }
+
+    /// @return 'true' if replica info has to be saved in a database
+    bool saveReplicaInfo() const { return _saveReplicaInfo; }
+
+    /**
+     * Return the result of the operation.
+     *
+     * IMPORTANT NOTES:
+     * - the method should be invoked only after the job has finished (primary
+     *   status is set to Job::Status::FINISHED). Otherwise exception
+     *   std::logic_error will be thrown
+     *
+     * - the result will be extracted from requests which have successfully
+     *   finished. Please, verify the primary and extended status of the object
+     *   to ensure that all requests have finished.
+     *
+     * @return the data structure to be filled upon the completin of the job.
+     *
+     * @throws std::logic_error - if the job dodn't finished at a time
```

Comment:
"isn't finished at the time when the method is called"?

Reply:
fixed